common/notification.ts

```diff
@@ -0,0 +1,29 @@
+export type Notification = {
+  id: string
+  userId: string
+  reasonText?: string
+  reason?: notification_reason_types
+  createdTime: number
+  viewTime?: number
+  isSeen: boolean
+
+  sourceId?: string
+  sourceType?: notification_source_types
+  sourceContractId?: string
+  sourceUserName?: string
+  sourceUserUsername?: string
+  sourceUserAvatarUrl?: string
+}
+export type notification_source_types =
+  | 'contract'
+  | 'comment'
+  | 'bet'
+  | 'answer'
+  | 'liquidity'
+
+export type notification_reason_types =
+  | 'created'
```

Comment:
The idea is to use the 'created' reason type with each of the different source types? That might make it slightly more annoying to query for new comment notifications b/c, I believe, Firestore will require generating a new index.

Reply:
The reason types are for front end generation of the text that the user will see, not for firestore querying - i.e. 'someone commented/edited a comment/deleted a comment etc.' . The query will just use isSeen.